tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,761 @@
+                    .build();
+            FileSystemRealmUtil.createEncryptedRealmFromUnencrypted(oldFileSystemRealm, newFileSystemRealm);
+        }
+    }
+
+    /**
+     * Creates the script/commands the user must run for Elytron to recognize
+     * and use the new filesystem-realm
+     */
+
+    private void createWildFlyScript() throws Exception {
+        int counter = 0;
+        for (Descriptor descriptor : descriptors) {
+            String outputRealmLocation = descriptor.getOutputRealmLocation();
+            String fileSystemRealmName = descriptor.getFileSystemRealmName();
+            String credentialStore = descriptor.getCredentialStore();
+            String secretKeyAlias = descriptor.getSecretKeyAlias();
+            int levels = descriptor.getLevels();
+
+            if(secretKeyAlias == null) {
+                secretKeyAlias = "key";
+            }
+            String createScriptCheck = "";
+            if (Paths.get(String.format("%s.sh", fileSystemRealmName)).toFile().exists()) {
+                createScriptCheck = prompt(
+                        false,
+                        null,
+                        false,
+                        ElytronToolMessages.msg.shouldFileBeOverwritten(String.format("%s.sh", fileSystemRealmName))
+                );
+            }
+            String fullOutputPath;
+            if (outputRealmLocation.startsWith(".")) {
+                fullOutputPath = Paths.get(outputRealmLocation.substring(2)).toAbsolutePath().toString();
+            } else {
+                fullOutputPath = Paths.get(outputRealmLocation).toAbsolutePath().toString();
+            }
+
+            if (summaryMode) {
+                summaryString.append(String.format("Configured script for WildFly named %s.sh at %s.", fileSystemRealmName, fullOutputPath));
+                summaryString.append(System.getProperty("line.separator"));
+                summaryString.append("The script is using the following names:");
+                summaryString.append(System.getProperty("line.separator"));
+                summaryString.append(String.format("Name of filesystem-realm: %s", fileSystemRealmName));
+                summaryString.append(System.getProperty("line.separator"));
+            }
+
+            List<String> scriptLines = Arrays.asList(
+                String.format("/subsystem=elytron/secret-key-credential-store=%s:add(path=%s)", "mycredstore"+counter, credentialStore),
+                String.format("/subsystem=elytron/filesystem-realm=%s:add(path=%s, levels=%s, credential-store=%s, secret-key=%s)", fileSystemRealmName, fullOutputPath+'/'+fileSystemRealmName, levels, "mycredstore"+counter, secretKeyAlias)
+            );
+
+            if (!createScriptCheck.equals("y") && !createScriptCheck.equals("yes")) {
+                Files.write(Paths.get(String.format("%s/%s.sh", outputRealmLocation, fileSystemRealmName)), scriptLines, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
```

Comment:
When the `outputRealmLocation` directory does not exist, this results into `NoSuchFileException`:

```
$ ./bin/elytron-tool.sh filesystem-realm-encrypt -i fs-realm-users-plain -o fs-realm-users-plain-enc --debug
Creating encrypted realm for: fs-realm-users-plain
Exception encountered executing the command:
java.nio.file.NoSuchFileException: fs-realm-users-plain-enc/encrypted-filesystem-realm.sh
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:220)
	at java.base/java.nio.file.Files.write(Files.java:3490)
	at java.base/java.nio.file.Files.write(Files.java:3541)
	at org.wildfly.security.elytron-tool@1.18.4.CR1-SNAPSHOT//org.wildfly.security.tool.FileSystemEncryptRealmCommand.createWildFlyScript(FileSystemEncryptRealmCommand.java:753)
	at org.wildfly.security.elytron-tool@1.18.4.CR1-SNAPSHOT//org.wildfly.security.tool.FileSystemEncryptRealmCommand.execute(FileSystemEncryptRealmCommand.java:388)
	at org.wildfly.security.elytron-tool@1.18.4.CR1-SNAPSHOT//org.wildfly.security.tool.ElytronTool.main(ElytronTool.java:85)
	at org.jboss.modules.Module.run(Module.java:353)
	at org.jboss.modules.Module.run(Module.java:321)
	at org.jboss.modules.Main.main(Main.java:604)
```

Reply:
Changed to now create `outputRealmLocation` directory if it does not exist
https://github.com/wildfly-security/wildfly-elytron/pull/1676/files#diff-ebe53241afd70e1e5743d6db5b39150461c335abce6c083cafbf6aae4b4c1e2dR329-R332